core/invoker/src/main/scala/whisk/core/containerpool/ContainerPool.scala

```diff
@@ -156,14 +156,29 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
             // (and a new container would over commit the pool)
             val isErrorLogged = r.retryLogDeadline.map(_.isOverdue).getOrElse(true)
             val retryLogDeadline = if (isErrorLogged) {
-              logging.error(
-                this,
+              def errorMessage(reason: String) =
                 s"Rescheduling Run message, too many message in the pool, " +
+                  reason +
                   s"freePoolSize: ${freePool.size} containers and ${memoryConsumptionOf(freePool)} MB, " +
                   s"busyPoolSize: ${busyPool.size} containers and ${memoryConsumptionOf(busyPool)} MB, " +
                   s"maxContainersMemory ${poolConfig.userMemory.toMB} MB, " +
                   s"userNamespace: ${r.msg.user.namespace.name}, action: ${r.action}, " +
-                  s"needed memory: ${r.action.limits.memory.megabytes} MB")(r.msg.transid)
+                  s"needed memory: ${r.action.limits.memory.megabytes} MB, " +
+                  s"waiting messages: ${runBuffer.size}"
+              if (runBuffer.size > 1) {
```

Comment:
Why not just emit the buffer size (always) instead of the explanation which may not be correct (plus too verbose) imo.